src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Com.cs

```diff
@@ -102,15 +102,35 @@ public static IntPtr GetIUnknownForObject(object o)
         }
 
         [SupportedOSPlatform("windows")]
-        public static void GetNativeVariantForObject(object? obj, IntPtr pDstNativeVariant)
+        public static unsafe void GetNativeVariantForObject(object? obj, IntPtr pDstNativeVariant)
         {
-            throw new PlatformNotSupportedException(SR.PlatformNotSupported_ComInterop);
+            Variant* data = (Variant*)pDstNativeVariant;
+            if (obj == null)
+            {
+                data->SetAsNULL();
+                return;
+            }
+
+            if (obj is byte)
```

Comment:
This should handle the same set of cases as the other direction.

Reply:
I was not sure. You may have idea in what order should I check each type based on usage. That was just a sketch.